lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -284,71 +284,70 @@ def filter_stream(tag, es)
     end
 
     def filter_stream_from_files(tag, es)
-      unless es.empty?
-        new_es = MultiEventStream.new
-
-        match_data = tag.match(@tag_to_kubernetes_name_regexp_compiled)
-        batch_miss_cache = {}
-        if match_data
-          container_id = match_data['docker_id']
-          metadata = {
-            'docker' => {
-              'container_id' => container_id
-            },
-            'kubernetes' => get_metadata_for_record(match_data, container_id, create_time_from_record(es.first[1]), batch_miss_cache)
-          }
-        end
+      if es.empty?
+        return es
+      end
+      new_es = MultiEventStream.new
+
+      match_data = tag.match(@tag_to_kubernetes_name_regexp_compiled)
+      batch_miss_cache = {}
+      if match_data
+        container_id = match_data['docker_id']
+        metadata = {
+          'docker' => {
+            'container_id' => container_id
+          },
+          'kubernetes' => get_metadata_for_record(match_data, container_id, create_time_from_record(es.first[1]), batch_miss_cache)
+        }
+      end
 
-        es.each { |time, record|
-          record = merge_json_log(record) if @merge_json_log
+      es.each { |time, record|
+        record = merge_json_log(record) if @merge_json_log
 
-          record = record.merge(Marshal.load(Marshal.dump(metadata))) if metadata
+        record = record.merge(Marshal.load(Marshal.dump(metadata))) if metadata
 
-          new_es.add(time, record)
-        }
-        dump_stats
-        new_es
-      end
+        new_es.add(time, record)
+      }
+      dump_stats
+      new_es
     end
 
     def filter_stream_from_journal(tag, es)
-      unless es.empty?
```

Comment:
We seem to have lost the empty check here

Reply:
We don't need it here - the code already correctly handles empty es.